home_vending_machines/api.lua

```diff
@@ -0,0 +1,59 @@
+local registered_currency = {}
+
+function home_vending_machines.register_currency(name, worth)
+    registered_currency[name] = worth
+end
+
+local function reg_simple(name, def)
+    minetest.register_node(":" .. name, {
+        description = def.description,
+        drawtype = "mesh",
+        mesh = "home_vending_machines_machine.obj",
+        tiles = def.tiles,
+        paramtype = "light",
+        paramtype2 = "facedir",
+        groups = def.groups or {snappy=3},
+        selection_box = {
+            type = "fixed",
+            fixed = {-0.5, -0.5, -0.5, 0.5, 1.5, 0.5}
+        },
+        collision_box = {
+            type = "fixed",
+            fixed = {-0.5, -0.5, -0.5, 0.5, 1.5, 0.5}
+        },
+        sounds = def.sounds,
+        on_rotate = function(pos, node, user, mode, new_param2)
+            if minetest.get_modpath("screwdriver") then
+                return screwdriver.rotate_simple(pos, node, user, mode, new_param2)
+            end
+        end,
+        on_rightclick = function(pos, node, clicker, itemstack, pointed_thing)
+            local pname = clicker:get_player_name()
+            local iname = itemstack:get_name()
+            local dpos = vector.add((vector.multiply(minetest.facedir_to_dir(node.param2), -1)), pos)
+
+            if registered_currency[iname] and registered_currency[iname] == 1 then
+                local item = def._vmachine.item
+                if type(item) == "table" then
+                    item = item[math.random(#item)]
+                end
+
+                minetest.spawn_item(dpos, item)
```

Comment:
Apparently `minetest.spawn_item` is an undocumented Lua-implemented version of `minetest.add_item`. 🤔

Reply:
yup

Reply:
probably should swap